src/pages/register/components/RoleLayout.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useState } from "react";
+import * as S from "../Register.styled";
+import Spacing from "@components/commons/spacing/Spacing";
+import { IconRoleAdd } from "@assets/svgs";
+import RoleWrapper from "./RoleWrapper";
+import { Cast, Staff } from "../typings/gigInfo";
+
+interface Role {
+  id: number;
+  makerName: string;
+  makerRole: string;
+  makerPhoto: string;
+}
+
+interface RoleLayoutProps {
+  list: Cast[] | Staff[];
+  updateList: (list: any[]) => void;
+  title: string;
+}
+
+const RoleLayout = ({ list, updateList, title }: RoleLayoutProps) => {
+  const [makerList, setMakerList] = useState<Role[]>(
+    list.map((item, index) => ({
+      id: index, // ID 생성
```

Comment:
p4) 아마 알고 계시겠지만,,, index는 변경 가능성이 있어 id값으로 주지 않는 게 좋다고 알고 있어서요!! id를 
```suggestion
      id: `${castName}-${index}`
```
이런 식으로 변경하는 건 어떨까요?

Reply:
id 타입이 숫자라서 제안해주신 코드로 변경하기는 어려울 것 같습니다...🥹 혹시 더 좋은 방법이 있으시다면 제안해주세요..!